lib/system/web/src/keyboard_engine.rs

```diff
@@ -0,0 +1,170 @@
+use std::collections::{HashMap, BTreeSet};
+use std::cell::RefCell;
+use std::rc::Rc;
+use wasm_bindgen::JsCast;
+use wasm_bindgen::prelude::*;
+use itertools::Itertools;
+
+enum Action {
+    KeyUp,
+    KeyDown
+}
+
+type Bindings = HashMap<String, Box<dyn FnMut()>>;
+
+#[wasm_bindgen]
+pub struct KeyboardEngine {
+    bindings:        Rc<RefCell<Bindings>>,
```

Comment:
Try not using internal mutability as far as you can. Of course, in our final application we might want to keep `KeyboardEngine` as internal mutable struct, but then we will keep `Rc<RefCell<KeyboardEngine>>`. Try not to use `RefCell` in **any place** in your code now.

Reply:
Do you have an idea how **to mutable** bindings field then?

Reply:
I'm using `RefCell` not for internal mutability but cause I need to have few owners and mutate the field at the same time (because of the js async nature). If I use just `Rc` I got an `None` on `Rc::get_mut` when have more then 1 reference. Maybe you know how to do it better?